Your Addresses' View

jessgusclark opened the pull request.

jessgusclark commented on src/app/tabs/newAdmin/addresses/networks.json:
```diff
@@ -0,0 +1,30 @@
+    "id": "0x80000091",
+    "icon": "/assets/icons/icon_bitconcash.png"
+  }
+]
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/operations.js:
```diff
@@ -0,0 +1,210 @@
+  const hash = namehash(domain);
+  const chainName = getChainNameById(chainId);
+
+  return new Promise((resolve) => {
+    multichainResolver.methods.chainAddr(hash, chainId).call((error, result) => {
+      if (error) {
+        return errorChainAddress(chainName, error.message);
+      }
+      return resolve(dispatch(receiveChainAddress(chainId, chainName, result)));
+    });
+  });
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/operations.js:
```diff
@@ -0,0 +1,210 @@
+  dispatch(requestChainAddress());
+  const hash = namehash(domain);
+
+  return new Promise((resolve) => {
+    publicResolver.methods.addr(hash).call((error, result) => {
+      if (error) {
+        return errorChainAddress('RSK', error.message);
+      }
+      return resolve(dispatch(receiveChainAddress('0x80000089', 'RSK', result)));
+    });
+  });
```
I change this to similar to the multichain (above), and removed the promise. So it is now:
```
return publicResolver.methods.addr(hash).call()
    .then(addr => dispatch(receiveChainAddress('0x80000089', 'RSK', addr)))
    .catch(error => dispatch(errorChainAddress('RSK', error.message)));
```

ilanolkies replied:
I yes. I forgot erasing the Promise :D

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/MigrateToMultiResolverComponent.js:
```diff
@@ -0,0 +1,64 @@
+          className="migrate"
+          disabled={isEditing}
+        >
+          {strings.migrate}
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+  <Row>
+    <Col>
+      <h1>
+        {strings.your_addresses}
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/MigrateToMultiResolverComponent.js:
```diff
@@ -0,0 +1,64 @@
+        message={strings.wait_transation_confirmed}
+        visible={isWaiting}
+      />
+      <UserErrorComponent
```
👍 Reset the error message when you click on "migrate".

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+        {strings.your_addresses}
+        {resolverName === MULTICHAIN_RESOLVER && ` - ${strings.multichain}`}
+      </h1>
+      <p>{strings.your_addresses_explanation}</p>
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+    <Col>
+      <h1>
+        {strings.your_addresses}
+        {resolverName === MULTICHAIN_RESOLVER && ` - ${strings.multichain}`}
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+              labelIcon={network.icon}
+              networkId={chainId}
+              value={address}
+              valueDisplay={isHex ? toChecksumAddress(address, networkChainId) : address}
```
👍 I also removed the nested div.checksumError.

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+                value_prefix: strings.value,
+                error_message: errorMessage,
+                cancel: strings.cancel,
+                submit: strings.submit,
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+  resolverName: propTypes.string.isRequired,
+};
+
+export default multilanguage(YourAddressesComponent);
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+  resolverName: propTypes.string.isRequired,
+};
+
+export default multilanguage(YourAddressesComponent);
```
👍 Clicking "convert to lowercase" will now execute the submit function and bring up the wallet. It has been going through the validation process, but since checking if it is a valid checksum is the last thing in the validation check, this is not needed.

ilanolkies replied:
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+  resolverName: propTypes.string.isRequired,
+};
+
+export default multilanguage(YourAddressesComponent);
```
The "success" box must be closed on the individual input before you can edit it again. Once that is closed, the component works as expected.

ilanolkies replied:
Is this the expected behaviour?

jessgusclark replied:
I would think yes, that you should finish up the action that you are working on before moving to the next.

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+  resolverName: propTypes.string.isRequired,
+};
+
+export default multilanguage(YourAddressesComponent);
```
👍 chainId should be set to null, not 1.

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/YourAddressesComponent.js:
```diff
@@ -0,0 +1,87 @@
+  resolverName: propTypes.string.isRequired,
+};
+
+export default multilanguage(YourAddressesComponent);
```
👍

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/AddNewAddressComponent.js:
```diff
@@ -0,0 +1,157 @@
+      <Row>
+        <Col>
+          <h2 className="gray normal-size">
+            {strings.add_new_chain_address}
```
👍 Added the gray style for headers

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/AddNewAddressComponent.js:
```diff
@@ -0,0 +1,157 @@
+      </Row>
+      <Row>
+        <Col md={3}>
+          <select
```
I'm assuming you are on a mac. What browser are you using?

Here is what I see in Windows: 

Chrome:
![image](https://user-images.githubusercontent.com/766679/78360991-51840a80-75c0-11ea-91d7-4238ecfae69f.png)

Firefox:
![image](https://user-images.githubusercontent.com/766679/78361024-5e086300-75c0-11ea-9298-a819ef8b8af2.png)

ilanolkies replied:
🦊 😖 let's leave default one for no one. Then we can create something custom and also add the icons there.

jessgusclark commented on src/app/tabs/newAdmin/addresses/components/AddNewAddressComponent.js:
```diff
@@ -0,0 +1,157 @@
+  const handleAddClick = () => {
+    const networkInfo = allNetworks.filter(net => net.id === selectedNetwork)[0];
+    if (networkInfo.validation && networkInfo.validation === 'HEX') {
+      const validationChainId = allNetworks.filter(net => net.id === selectedNetwork)[0].checksum
```
👍 I changed this to always be `null` except when using an RSK address. There, it will pick it up from the env file.

jessgusclark commented:
@ilanolkies 

> * Can we set the same height for the coin selector and the address input?

You mentioned that you are on a mac with Firefox, correct? Can you send me a picture of what this looks like on your device? This is what it looks like on my device:

chrome:
![image](https://user-images.githubusercontent.com/766679/78772942-2bcd7b80-799b-11ea-8826-4f662ece0a1e.png)

firefox:
![image](https://user-images.githubusercontent.com/766679/78772841-09d3f900-799b-11ea-91da-1fd931e74ecf.png)

> * Change 'add new **chain** address' for 'add new address'

Updated.

> * When I have no multi chain resolver, the checksum for rsk addresses is wrong. It throws the react error screen when createing the value or updating the value. Try setting '0x82c6febb170EE10a019F434C6e4337EA3fcaA6AB' that is valid for testnet.

Fixed this, it wasn't converting to lowercase before sending it to the contract.

ilanolkies commented:
Great job!
